### PR TITLE
New Feature: Implemented Pagination in each unit "GetAll()"s

### PR DESCRIPTION
**Pull request by Ozodbek00**

### CODE REVIEW

**Comment by Ozodbek00** [comment]
@shurik86

File: iTechArt.Api/Controllers/AirportController.cs [hunk 1]
```diff
@@ -44,9 +44,9 @@ public async Task<IActionResult> ImportAirportExcel(IFormFile file)
         /// Controller of Exporting airport data
         /// </summary>
         [HttpGet("get_all")]
-        public async Task<IActionResult> ExportAirportExcel()
+        public async Task<IActionResult> ExportAirportExcel(int pageIndex)
```
Comment (shurik86) [inline iTechArt.Api/Controllers/AirportController.cs]: difference methods in airport and grocery, what way is right?

Comment (shurik86) [reply iTechArt.Api/Controllers/AirportController.cs]: async method, by name doesn`t contain 'Async' postfix

File: iTechArt.Api/Controllers/GroceryController.cs [hunk 2]
```diff
@@ -89,9 +89,9 @@ public async ValueTask<IActionResult> ImportXMLGroceryFileAsync(IFormFile file)
         /// api route which allows to get all info from db and parse it to the following format
         /// </summary>
         [HttpGet("get_all")]
-        public async Task<IActionResult> ExportGroceryDataAsync()
+        public async Task<IActionResult> ExportGroceryData([FromQuery] int pageIndex)
```
Comment (shurik86) [inline iTechArt.Api/Controllers/GroceryController.cs]: Async method doesn`t contain 'Async' postfix.
Check another methods in project

File: iTechArt.Api/Controllers/StudentsController.cs [hunk 3]
```diff
@@ -34,22 +34,18 @@ public async Task<ActionResult> ImportAsync(IFormFile file)
...
+             
+            return BadRequest("Invalid file format!");
         }
 
         /// <summary>
         /// Get all students
         /// </summary>
         [HttpGet("get_all")]
-        public async Task<ActionResult<IStudent[]>> GetAllAsync()
+        public async Task<ActionResult<IStudent[]>> GetAllAsync([FromQuery] int pageIndex)
```
Comment (shurik86) [inline iTechArt.Api/Controllers/StudentsController.cs:46]: if UI what to change paging count elements, they are can`t do it.

File: iTechArt.Repository/PaginationHelpers/Pagination.cs [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+{
+    internal static class Pagination
+    {
+        /// <summary>
+        /// Paginates DbSet according to the pageIndex and PageSize
+        /// </summary>
+        public static IQueryable<TSource> Paginate<TSource>(this IQueryable<TSource> source, int pageIndex)
+        {
+            return PaginationConstant.PageSize > 0 && pageIndex > 0
+                ? source.Skip((pageIndex - 1) * PaginationConstant.PageSize).Take(PaginationConstant.PageSize)
```
Comment (shurik86) [inline iTechArt.Repository/PaginationHelpers/Pagination.cs]: PaginationConstant.PageSize it is constant, why you compare it with 0?

File: iTechArt.Repository/Repositories/GroceryRepository.cs [hunk 5]
```diff
@@ -20,9 +21,9 @@ public GroceryRepository(AppDbContext dbContext, IMapper mapper)
         /// <summary>
         /// Get all entities from database.
         /// </summary>
-        public async Task<IGrocery[]> GetAllAsync()
+        public async Task<IGrocery[]> GetAllAsync(int pageIndex)
         {
-            return await _dbContext.Groceries.Select(grocery => _mapper.Map<Grocery>(grocery)).ToArrayAsync();
+            return await _dbContext.Groceries.Paginate(pageIndex).Select(groceries => _mapper.Map<Grocery>(groceries)).ToArrayAsync();
```
Comment (shurik86) [inline iTechArt.Repository/Repositories/GroceryRepository.cs:26]: groceries means many of grocery.
But in select statement it is one element.

**Review by shurik86 — COMMENTED** [review]
Fix review issues please

File: iTechArt.Api/Controllers/AirportController.cs (hunk 1 above)
Comment (Ozodbek00) [reply iTechArt.Api/Controllers/AirportController.cs]: Changed all task returning methods into Async()

File: iTechArt.Api/Controllers/GroceryController.cs (hunk 2 above)
Comment (Ozodbek00) [reply iTechArt.Api/Controllers/GroceryController.cs]: Done

File: iTechArt.Api/Controllers/StudentsController.cs (hunk 3 above)
Comment (Ozodbek00) [reply iTechArt.Api/Controllers/StudentsController.cs:46]: We decided to make it constant
If we need, we may make some changes

File: iTechArt.Repository/PaginationHelpers/Pagination.cs (hunk 4 above)
Comment (Ozodbek00) [reply iTechArt.Repository/PaginationHelpers/Pagination.cs]: done

File: iTechArt.Repository/Repositories/GroceryRepository.cs (hunk 5 above)
Comment (Ozodbek00) [reply iTechArt.Repository/Repositories/GroceryRepository.cs:26]: It is not my unit so did not make serious changes in structure

Comment (shurik86) [reply iTechArt.Repository/Repositories/GroceryRepository.cs:26]: you rename grocery to groceries

**Pull request merged**